test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1910 @@
+          expectedDistUniEif),
+    mergeToRow(unordered_rows_small.at(4), unordered_rows_small.at(1),
+          expectedDistMunEif),
+    mergeToRow(unordered_rows_small.at(4), unordered_rows_small.at(2),
+          expectedDistEyeEif)
+  };
+
+  std::vector<std::vector<std::string>> expectedMaxDist10000000_rows_small{
+    mergeToRow(unordered_rows_small.at(0), unordered_rows_small.at(0),
+          expectedDistSelf),
+    mergeToRow(unordered_rows_small.at(0), unordered_rows_small.at(1),
+          expectedDistUniMun),
+    mergeToRow(unordered_rows_small.at(0), unordered_rows_small.at(4),
+          expectedDistUniEif),
+    mergeToRow(unordered_rows_small.at(0), unordered_rows_small.at(2),
+          expectedDistUniEye),
+    mergeToRow(unordered_rows_small.at(0), unordered_rows_small.at(3),
+          expectedDistUniLib),
+    mergeToRow(unordered_rows_small.at(1), unordered_rows_small.at(1),
+          expectedDistSelf),
+    mergeToRow(unordered_rows_small.at(1), unordered_rows_small.at(0),
+          expectedDistUniMun),
+    mergeToRow(unordered_rows_small.at(1), unordered_rows_small.at(4),
+          expectedDistMunEif),
+    mergeToRow(unordered_rows_small.at(1), unordered_rows_small.at(2),
+          expectedDistMunEye),
+    mergeToRow(unordered_rows_small.at(1), unordered_rows_small.at(3),
+          expectedDistMunLib),
+    mergeToRow(unordered_rows_small.at(2), unordered_rows_small.at(2),
+          expectedDistSelf),
+    mergeToRow(unordered_rows_small.at(2), unordered_rows_small.at(4),
+          expectedDistEyeEif),
+    mergeToRow(unordered_rows_small.at(2), unordered_rows_small.at(0),
+          expectedDistUniEye),
+    mergeToRow(unordered_rows_small.at(2), unordered_rows_small.at(1),
+          expectedDistMunEye),
+    mergeToRow(unordered_rows_small.at(2), unordered_rows_small.at(3),
+          expectedDistEyeLib),
+    mergeToRow(unordered_rows_small.at(3), unordered_rows_small.at(3),
+          expectedDistSelf),
+    mergeToRow(unordered_rows_small.at(3), unordered_rows_small.at(0),
+          expectedDistUniLib),
+    mergeToRow(unordered_rows_small.at(3), unordered_rows_small.at(1),
+          expectedDistMunLib),
+    mergeToRow(unordered_rows_small.at(3), unordered_rows_small.at(2),
+          expectedDistEyeLib),
+    mergeToRow(unordered_rows_small.at(3), unordered_rows_small.at(4),
+          expectedDistEifLib),
+    mergeToRow(unordered_rows_small.at(4), unordered_rows_small.at(4),
+          expectedDistSelf),
+    mergeToRow(unordered_rows_small.at(4), unordered_rows_small.at(0),
+          expectedDistUniEif),
+    mergeToRow(unordered_rows_small.at(4), unordered_rows_small.at(1),
+          expectedDistMunEif),
+    mergeToRow(unordered_rows_small.at(4), unordered_rows_small.at(2),
+          expectedDistEyeEif),
+    mergeToRow(unordered_rows_small.at(4), unordered_rows_small.at(3),
+          expectedDistEifLib)
+  };
+
+  std::vector<std::vector<std::string>> expectedMaxDist1_rows_diff{
```

Comment:
This is not as straight forward as it may seem, because it's triples (row1, row2, distance). Therefore a pair implementation does not work. Should i check, if the code could be refactored in another way?